src/PropertyHandler/ObjectExporter.php

```diff
@@ -122,4 +125,17 @@ public function canExport(Field $field, mixed $value, string $format): bool
     {
         return $field->typeCategory === TypeCategory::Object;
     }
+
+    private function getPropertyReader(object $value): callable
+    {
+        return (static function (string $prop) use ($value): mixed {
+            // get property value using reflection
+            $prop = new \ReflectionProperty($value, $prop);
+            if (!$prop->isInitialized($value)) {
+                return SerdeError::Uninitialized;
+            }
+            $prop->setAccessible(true);
+            return $prop->getValue($value);
+        });
+    }
```

Comment:
The previous variant was so elegant, yet I was not unable to properly distinguish between `null` and `uninitialized`.

Reply:
I'm concerned about the performance impact here.  Reflection is not the slowest API in PHP, but it's not the fastest, either, AFAIK.  This is adding a new reflection object creation for every property in the entire tree being serialized.  That's something that needs to be benchmarked.

The setAccessible() call is also unnecessary.  Binding the function to the object being read is cleaner and faster, as it doesn't require modifying the original object.

Reply:
Ok, after some fiddling with OpenAI and Copilot I was able to find a way how to keep the binded closure by using `get_object_vars`, if there is a better way please hint.

Reply:
The first two conditions seem redundant.  Wouldn't anything that passes `isset()` also pass `array_key_exists()`?  That would also then allow the code to be condensed down to a single ternary statement again, I think.